test/new-e2e/tests/otel/utils/docker_pipelines_utils.go

```diff
@@ -0,0 +1,154 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package utils
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/DataDog/datadog-agent/test/fakeintake/aggregator"
+	fakeintake "github.com/DataDog/datadog-agent/test/fakeintake/client"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/environments"
+)
+
+// OTelDockerTestSuite is an interface for the OTel e2e test suite.
+type OTelDockerTestSuite interface {
+	T() *testing.T
+	Env() *environments.DockerHost
+}
+
+// TestCalendarAppDocker verifies the calendar app is sending telemetry for e2e tests
+func TestCalendarAppDocker(s OTelDockerTestSuite) {
+	err := s.Env().FakeIntake.Client().FlushServerAndResetAggregators()
+	require.NoError(s.T(), err)
+
+	s.T().Log("Waiting for calendar app to start")
+	// Wait for calendar app to start
+	require.EventuallyWithT(s.T(), func(c *assert.CollectT) {
+		logs, err := s.Env().FakeIntake.Client().FilterLogs(calendarService, fakeintake.WithMessageContaining(logBody))
+		assert.NoError(c, err)
+		assert.NotEmpty(c, logs)
+	}, 30*time.Minute, 10*time.Second)
+}
+
+// TestTracesDocker tests that OTLP traces are received through OTel pipelines as expected
+func TestTracesDocker(s OTelDockerTestSuite) {
```

Comment:
Curious what are the differences between these funcs and the ones in pipelines_utils, other than container tags? If most parts are the same I'd prefer to refactor and reuse pipelines_utils

Reply:
Container tags are the main difference, but the test suite type is also different between kubernetes and docker (`environments.DockerHost` vs `environments.Kubernetes`). I'm not sure if there would be a clean way to refactor this?

Reply:
Ack, not a big deal, if it's not trivial to refactor I'm ok with leaving it as is